javascript/app.js

```diff
@@ -148,12 +148,48 @@ cocktailApp.revealButton.addEventListener("click", function () {
         .then((response) => {
             return response.json();
         }).then((jsonResult) => {
-            const newArray = Object.keys(jsonResult.drinks[0]);
+            const drinkDetails = jsonResult.drinks[0];
+            const newArray = Object.keys(drinkDetails);
             console.log(jsonResult.drinks[0]);
-            cocktailApp.parseArray(newArray, cocktailApp.ingredients, "strIngredient", jsonResult.drinks[0]);
-            cocktailApp.parseArray(newArray, cocktailApp.measurements, "strMeasure", jsonResult.drinks[0]);
+
+            // Reset the ingredients and measurements arrays
+            cocktailApp.ingredients = [];
+            cocktailApp.measurements = [];
+
+            cocktailApp.parseArray(newArray, cocktailApp.ingredients, "strIngredient", drinkDetails);
+            cocktailApp.parseArray(newArray, cocktailApp.measurements, "strMeasure", drinkDetails);
             // print instructions to page
+            document.querySelector('.instructionList').textContent = drinkDetails.strInstructions;
             // print ingredients & measurements to page
+
+            const ingredientList = document.querySelector(".ingredientList");
+
+            // Reset the ingredient list
+            while (ingredientList.firstChild) {
```

Comment:
I really like this - I wouldn't have though of using .firstChild. Really cool.

Not necessary, but since we're using this method twice (here and in the other button) maybe we make it a method and just call it on both occasions?

Reply:
I found the .firstchild thing on StackOverflow.
Good call about creating a method!